core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -156,13 +160,42 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
 
     // Update on the server.
     this.workflowExecutionsService
-      .setIsBookmarked(this.workflow.wid, row.eId, !wasPreviouslyBookmarked)
+      .groupSetIsBookmarked(this.workflow.wid, [row.eId], wasPreviouslyBookmarked)
       .pipe(untilDestroyed(this))
       .subscribe({
         error: (_: unknown) => (row.bookmarked = wasPreviouslyBookmarked),
       });
   }
 
+  allBookmarkCheck(rows: Set<WorkflowExecutionsEntry>): boolean {
+    if (rows !== undefined) {
+      for (let row of rows) {
+        if (!row.bookmarked) {
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  setBookmarked(rows: Set<WorkflowExecutionsEntry>): void {
+    if (this.workflow.wid === undefined) return;
+    if (rows !== undefined) {
+      let eIds: number[] = [];
+      // update the bookmark locally
+      let isBookmarked = this.allBookmarkCheck(rows);
+      for (let row of rows.values()) {
```

Comment:
Please use .map to transform your array from one type to the other to avoid this loop